nncf/experimental/post_training/api/dataloader.py

```diff
@@ -0,0 +1,30 @@
+from typing import Tuple
+from typing import TypeVar
+
+from abc import ABC
+from abc import abstractmethod
+
+ModelInput = TypeVar('ModelInput')
+Target = TypeVar('Target')
+
+
+class DataLoader(ABC):
+    """
+    Base class provides interface to get elements of the dataset.
+    """
+
+    def __init__(self, batch_size=1, shuffle: bool = True):
+        self.batch_size = batch_size
+        self.shuffle = shuffle
```

Comment:
Why do we need to have a `shuffle` parameter for the DataLoader in the case we have RandomBatchSampler already?

Reply:
`shuffle` as well as `batch_size` parameters are needed to pass for `Sampler` creation. The interface doesn't assume that user would interact with Sampler.
The `Sampler `is created by `StatisticsCollector` based on the parameters in `Dataloader`.
Take a look at https://github.com/openvinotoolkit/nncf/pull/1123/files#diff-f11bed5a94590184f22f709a4f4ad531d2e8cb66c91cc4a1d1f53cbdaa826e40R29

Reply:
@KodiaqQ Are you satisfied?